p2p/src/utils/requester.rs

```diff
@@ -0,0 +1,348 @@
+//! Make concurrent requests to peers limited by rate and prioritized by performance.
+
+use commonware_cryptography::{PublicKey, Scheme};
+use commonware_runtime::Clock;
+use commonware_utils::PrioritySet;
+use governor::{
+    clock::Clock as GClock, middleware::NoOpMiddleware, state::keyed::HashMapStateStore, Quota,
+    RateLimiter,
+};
+use std::{
+    collections::{BTreeMap, BTreeSet, HashMap, HashSet},
+    time::{Duration, SystemTime},
+};
+
+/// Unique identifier for a request.
+///
+/// Once u64 requests have been made, the ID wraps around (resetting to zero).
+/// As long as there are less than u64 requests outstanding, this should not be
+/// an issue.
+pub type ID = u64;
+
+/// Configuration for `Requester`.
+pub struct Config<C: Scheme> {
+    /// Cryptographic primitives.
+    pub crypto: C,
+
+    /// Rate limit for requests per participant.
+    pub rate_limit: Quota,
+
+    /// Initial expected performance for new participants.
+    pub initial: Duration,
+
+    /// Timeout for requests.
+    pub timeout: Duration,
+}
+
+/// Send rate-limited requests to peers prioritized by performance.
+pub struct Requester<E: Clock + GClock, C: Scheme> {
```

Comment:
would be cool to have small comments for each field (could be `//` instead of `///` to orient the reader)